Update What's New (#2899)
surveyjs/survey-creator: docs/Creator-V2-Whats-New.md

The new Survey Creator introduces a native implementation for React (more on this below). Implementations for Angular and Vue are planned for future development. For those customers who still use Knockout in their applications, we also ship an updated Survey Creator for Knockout with new look and feel that match the React version.

<a id="react-version"></a>

## Native Survey Creator for React

New Survey Creator for React is built from true React components. The native implementation became possible because we separated the platform-independent model code (now distributed as the [`survey-creator-core`](https://github.com/surveyjs/survey-creator/tree/master/packages/survey-creator-core) package) from the platform-specific rendering code (the [`survey-creator-react`](https://github.com/surveyjs/survey-creator/tree/master/packages/survey-creator-react) package). You can override the default rendering functions of our React components and integrate your own React components (view the [Override the Property Grid Component](https://surveyjs.io/Examples/Survey-Creator?id=overridepropertygrid&platform=ReactjsV2) example). We implemented the same concept in the SurveyJS Library earlier.

> Most of the UI elements in the new Survey Creator (Property Grid, Translation and Logic Tabs, modal editors) are stylized surveys from our own SurveyJS Library for React. This feature allows you to integrate any 3rd-party React component into both Survey Creator and SurveyJS library. View examples that show how to integrate the [React Select](https://react-select.com/home) component [into a standalone survey](https://surveyjs.io/Examples/Library?id=custom-widget-react-select) or [into the Survey Creator](https://surveyjs.io/Examples/Survey-Creator?id=react-select).

Reviewer: I'm not sure I get this. Let's say our Property Grid is not a stylized SurveyJS survey, but instead it's some third-party component. Would that mean I cannot integrated the React Select component into a standalone survey? 

The logic seems weird. Our elements are stylized surveys, hence you can integrate React components into our UI.

<a id="ui-changes"></a>

## Major UI/UX changes

> We redesigned most parts of the Survey Creator to make its UI more adaptive and UX more intuitive. Refer to help topics in this section for details.

Reviewer: Refer to help topics? Maybe "refer to new feature descriptions in this section for details"?

<a id="toolbox-and-property-grid"></a>

### New Toolbox and Property Grid design

> In the new design, the Toolbox and Property Grid look modern and adapt themselves to different layouts.

Reviewer: "In the new design" sounds Russian to me :-) 

```suggestion
Re-designed Toolbox and Property Grid look modern and adapt themselves to different layouts.
```

<img src="./images/creator-v2-toolbox-and-property-grid.png" alt="Survey Creator - Toolbox and Property Grid" width="75%">

<a id="action-buttons"></a> 

### Action buttons relocated

> Action buttons used to be displayed above the question you configure. Now, these buttons are displayed within the question and can be accessed programmatically.

Reviewer: Passive...

```suggestion
We now display action buttons within their corresponding question (instead of above the question). Programmatic access to the buttons is now possible.
```

<a id="logic-tab"></a>

### Logic tab redesigned

Users can now create and edit logic rules within the same view.

<img src="./images/creator-v2-logic-tab-item-editing.png" alt="Survey Creator - Logic tab" width="75%">

We also added a capability to filter the rules by questions or action types. This capability is useful for surveys with complex logic.

<img src="./images/creator-v2-logic-tab-filtering.png" alt="Survey Creator - Question selector in the Logic tab" width="75%">

<a id="test-tab"></a>

### Preview survey on different devices

We renamed the Test Survey tab to Preview and added a device selector that allows users to preview the created survey on most popular devices.

<img src="./images/creator-v2-device-type.png" alt="Survey Creator - Preview tab" width="75%">

<a id="technical-changes"></a>

## Technical changes

### Survey Creator UI elements are surveys

In the new Survey Creator, we use our own SurveyJS Library to render most of the UI elements. The main benefit of this approach is that we didn't have to implement native rendering because the SurveyJS Library already supports it for all frameworks.

> This new internal design also prompted us to extend the Library with new features that were demanded by the Survey Creator. For example, question titles can now contain [context actions](https://surveyjs.io/Examples/Library?id=survey-titleactions), and [Comment](https://surveyjs.io/Documentation/Library?id=questioncommentmodel) questions can [automatically grow or shrink](https://surveyjs.io/Documentation/Library?id=questioncommentmodel#autoGrow) the input field to accommodate the content.

Reviewer: ```suggestion
This new internal design also prompted us to extend the Library with new features used in the Survey Creator. For example, question titles can now contain [context actions](https://surveyjs.io/Examples/Library?id=survey-titleactions), and [Comment](https://surveyjs.io/Documentation/Library?id=questioncommentmodel) questions can [automatically grow or shrink](https://surveyjs.io/Documentation/Library?id=questioncommentmodel#autoGrow) the input field to accommodate the content.
```

Since Survey Creator UI elements are surveys, you can customize them as you would customize surveys. For example, the Property Grid is a one-page survey in which every property is a question. To introduce a new or override an existing property editor, you need to define a custom question JSON configuration and implement functions that survey events call internally.

[View Custom Property Editor example](https://surveyjs.io/Examples/Survey-Creator?id=custompropertyeditor&platform=ReactjsV2#content-js)

### Lazy Rendering

To improve performance on large multi-page surveys, we introduce support for Lazy Rendering to our SurveyJS Library. This feature allows the survey to render questions outside the viewport as skeletons instead of rendering them in full.

[View Lazy Rendering example](https://surveyjs.io/Examples/Library?id=survey-lazy)

### Alternative Rendering for Adorners

> Previously, we added [adorners](https://surveyjs.io/Documentation/Survey-Creator?id=survey-customization#specify-adorner-availability) to survey elements _after_ the elements were rendered. This wasn't a very effective solution because we modified the DOM twice.

Reviewer: ```suggestion
Previously, we used to render elements first, and only then added [adorners](https://surveyjs.io/Documentation/Survey-Creator?id=survey-customization#specify-adorner-availability). This wasn't a very effective solution because we modified the DOM twice.
```

> In the new Survey Creator, we registered new rendering functions for survey elements. These functions override the default rendering functions. For example, to implement in-place text editing in the Designer, we render an element that supports editable content instead of the default string element.

Reviewer: I was expecting this paragraph to continue the thought about adorners. Something like "the new version only modifies the DOM once and adds adorners right away". But this text seems not to be related to adorners.
